std/math.d

```diff
@@ -2100,13 +2116,24 @@ L_was_nan:
         enum real OF =  16384.0L;
         enum real UF = -16382.0L;
 
-        // Special cases.
+        // Special cases. Raises an overflow or underflow flag accordingly,
+        // except in the case for CTFE, where there are no hardware controls.
         if (isNaN(x))
             return x;
         if (x > OF)
-            return real.infinity;
+        {
+            if (__ctfe)
+                return real.infinity;
+            else
+                return real.max * copysign(real.max, x);
+        }
         if (x < UF)
-            return 0.0;
+        {
+            if (__ctfe)
+                return 0.0;
+            else
+                return real.min_normal * copysign(real.min_normal, x);
```

Comment:
CTFE code returns `+0.0`, but runtime `-0.0`. This is a bad practice for math.
Please check other signs too.